public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx

```diff
@@ -1,85 +1,63 @@
-import React, { PureComponent } from 'react';
+import React, { memo, useState, useEffect } from 'react';
 
 // Types
 import { ExploreQueryFieldProps } from '@grafana/data';
-import { FormLabel } from '@grafana/ui';
 
 import { PrometheusDatasource } from '../datasource';
 import { PromQuery, PromOptions } from '../types';
 
 import PromQueryField from './PromQueryField';
-export type Props = ExploreQueryFieldProps<PrometheusDatasource, PromQuery, PromOptions>;
-
-interface State {
-  interval: string;
-}
+import { PromExploreExtraField } from './PromExploreExtraField';
 
-export class PromExploreQueryEditor extends PureComponent<Props, State> {
-  // Query target to be modified and used for queries
-  query: PromQuery;
+export type Props = ExploreQueryFieldProps<PrometheusDatasource, PromQuery, PromOptions>;
 
-  constructor(props: Props) {
-    super(props);
-    const { query } = props;
-    this.query = query;
-    // Query target properties that are fully controlled inputs
-    this.state = {
-      // Fully controlled text inputs
-      interval: query.interval,
-    };
+export const PromExploreQueryEditor = memo(function PromExploreQueryEditor(props: Props) {
+  const { query, data, datasource, history, onChange, onRunQuery } = props;
+  const [step, setStep] = useState(query.interval);
+
+  function onChangeQueryStep(value: string, override?: boolean) {
```

Comment:
`override` does not seem to be ever set to `true`. Seems like `onRunQuery` won't ever be called.

Reply:
Right; it's a victim of a copypaste from `PromQueryField`. Fixed.